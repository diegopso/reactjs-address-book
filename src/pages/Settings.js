import React from 'react'
import { Layout, PageHeader, Breadcrumb, Divider, List, Typography, Switch } from 'antd'
import { HomeOutlined, FlagTwoTone } from '@ant-design/icons'
import { MinimalHeader } from '../components'
import Flag from 'react-flagkit'

const nationalities = [
  { code: 'CH', name: 'Switzerland' },
  { code: 'ES', name: 'Spain' },
  { code: 'FR', name: 'France' },
  { code: 'GB', name: 'United Kingdom' }
]

const Settings = () => {
  return (
    <div>
      <MinimalHeader></MinimalHeader>
      <Layout className="internal-container">
        <Breadcrumb>
          <Breadcrumb.Item href="/">
            <HomeOutlined />
          </Breadcrumb.Item>
          <Breadcrumb.Item>Settings</Breadcrumb.Item>
        </Breadcrumb>
        <PageHeader
          onBack={() => window.history.back()}
          className="site-page-header"
          title="Settings"
        />
        <div>
          <Divider orientation="left"><FlagTwoTone twoToneColor="#B398DB" /> Nationalities</Divider>
          <List
            dataSource={nationalities}
            renderItem={item => (
              <List.Item>
                <Switch defaultChecked /> <Flag country={ item.code } size="24" /> <Typography.Text>{item.name}</Typography.Text>
              </List.Item>
            )}
          />
        </div>
      </Layout>
    </div>
  )
}

export default Settings
