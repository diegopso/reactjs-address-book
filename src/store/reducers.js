import { combineReducers } from 'redux'
import settingsReducer from './settings/reducers'
import paginationReducer from './pagination/reducers'

export default combineReducers({
  settings: settingsReducer,
  pagination: paginationReducer
})
