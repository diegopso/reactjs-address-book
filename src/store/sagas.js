import { all } from 'redux-saga/effects'
import paginationSaga from './pagination/sagas'
import toastSaga from './toast/sagas'

export default function * rootSaga (getState) {
  yield all([
    paginationSaga(),
    toastSaga()
  ])
}
