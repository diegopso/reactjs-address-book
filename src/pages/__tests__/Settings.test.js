import React from 'react'
import { describe, it, expect } from '@jest/globals'
import Settings from '../Settings'
import { Provider } from 'react-redux'
import { render } from '@testing-library/react'
import store from '../../store'
import { BrowserRouter as Router } from 'react-router-dom'

describe('Testing Settings component:', () => {
  it('Renders correctly.', () => {
    const wrapper = render(
      <Router>
        <Provider store={store}>
          <Settings />
        </Provider>
      </Router>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
