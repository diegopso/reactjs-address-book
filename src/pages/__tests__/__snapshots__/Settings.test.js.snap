// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Settings component: Renders correctly. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <header
          class="minimal-header"
        />
        <section
          class="ant-layout internal-container"
        >
          <div
            class="ant-breadcrumb"
          >
            <span>
              <a
                class="ant-breadcrumb-link"
                href="/"
              >
                <span
                  aria-label="home"
                  class="anticon anticon-home"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="home"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                    />
                  </svg>
                </span>
              </a>
              <span
                class="ant-breadcrumb-separator"
              >
                /
              </span>
            </span>
            <span>
              <span
                class="ant-breadcrumb-link"
              >
                Settings
              </span>
              <span
                class="ant-breadcrumb-separator"
              >
                /
              </span>
            </span>
          </div>
          <div
            class="ant-page-header site-page-header ant-page-header-ghost"
          >
            <div
              class="ant-page-header-heading"
            >
              <div
                class="ant-page-header-heading-left"
              >
                <div
                  class="ant-page-header-back"
                >
                  <div
                    aria-label="Back"
                    class="ant-page-header-back-button"
                    role="button"
                    style="border: 0px; background: transparent; padding: 0px; line-height: inherit; display: inline-block;"
                    tabindex="0"
                  >
                    <span
                      aria-label="arrow-left"
                      class="anticon anticon-arrow-left"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="arrow-left"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 000 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
                <span
                  class="ant-page-header-heading-title"
                  title="Settings"
                >
                  Settings
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-left"
              role="separator"
            >
              <span
                class="ant-divider-inner-text"
              >
                <span
                  aria-label="flag"
                  class="anticon anticon-flag"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="flag"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M184 232h368v336H184z"
                      fill="#f8f0ff"
                    />
                    <path
                      d="M624 632c0 4.4-3.6 8-8 8H504v73h336V377H624v255z"
                      fill="#f8f0ff"
                    />
                    <path
                      d="M880 305H624V192c0-17.7-14.3-32-32-32H184v-40c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v784c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V640h248v113c0 17.7 14.3 32 32 32h416c17.7 0 32-14.3 32-32V337c0-17.7-14.3-32-32-32zM184 568V232h368v336H184zm656 145H504v-73h112c4.4 0 8-3.6 8-8V377h216v336z"
                      fill="#B398DB"
                    />
                  </svg>
                </span>
                 Nationalities
              </span>
            </div>
            <div
              class="ant-list ant-list-split"
            >
              <div
                class="ant-spin-nested-loading"
              >
                <div
                  class="ant-spin-container"
                >
                  <ul
                    class="ant-list-items"
                  >
                    <li
                      class="ant-list-item ant-list-item-no-flex"
                    >
                      <button
                        aria-checked="true"
                        class="ant-switch ant-switch-checked"
                        role="switch"
                        type="button"
                      >
                        <div
                          class="ant-switch-handle"
                        />
                        <span
                          class="ant-switch-inner"
                        />
                      </button>
                       
                      <img
                        alt="CH Flag"
                        height="24"
                        role="img"
                        src="https://cdn.jsdelivr.net/gh/madebybowtie/FlagKit@2.2/Assets/SVG/CH.svg"
                        width="24"
                      />
                       
                      <span
                        class="ant-typography"
                      >
                        Switzerland
                      </span>
                    </li>
                    <li
                      class="ant-list-item ant-list-item-no-flex"
                    >
                      <button
                        aria-checked="true"
                        class="ant-switch ant-switch-checked"
                        role="switch"
                        type="button"
                      >
                        <div
                          class="ant-switch-handle"
                        />
                        <span
                          class="ant-switch-inner"
                        />
                      </button>
                       
                      <img
                        alt="ES Flag"
                        height="24"
                        role="img"
                        src="https://cdn.jsdelivr.net/gh/madebybowtie/FlagKit@2.2/Assets/SVG/ES.svg"
                        width="24"
                      />
                       
                      <span
                        class="ant-typography"
                      >
                        Spain
                      </span>
                    </li>
                    <li
                      class="ant-list-item ant-list-item-no-flex"
                    >
                      <button
                        aria-checked="true"
                        class="ant-switch ant-switch-checked"
                        role="switch"
                        type="button"
                      >
                        <div
                          class="ant-switch-handle"
                        />
                        <span
                          class="ant-switch-inner"
                        />
                      </button>
                       
                      <img
                        alt="FR Flag"
                        height="24"
                        role="img"
                        src="https://cdn.jsdelivr.net/gh/madebybowtie/FlagKit@2.2/Assets/SVG/FR.svg"
                        width="24"
                      />
                       
                      <span
                        class="ant-typography"
                      >
                        France
                      </span>
                    </li>
                    <li
                      class="ant-list-item ant-list-item-no-flex"
                    >
                      <button
                        aria-checked="true"
                        class="ant-switch ant-switch-checked"
                        role="switch"
                        type="button"
                      >
                        <div
                          class="ant-switch-handle"
                        />
                        <span
                          class="ant-switch-inner"
                        />
                      </button>
                       
                      <img
                        alt="GB Flag"
                        height="24"
                        role="img"
                        src="https://cdn.jsdelivr.net/gh/madebybowtie/FlagKit@2.2/Assets/SVG/GB.svg"
                        width="24"
                      />
                       
                      <span
                        class="ant-typography"
                      >
                        United Kingdom
                      </span>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <header
        class="minimal-header"
      />
      <section
        class="ant-layout internal-container"
      >
        <div
          class="ant-breadcrumb"
        >
          <span>
            <a
              class="ant-breadcrumb-link"
              href="/"
            >
              <span
                aria-label="home"
                class="anticon anticon-home"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="home"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                  />
                </svg>
              </span>
            </a>
            <span
              class="ant-breadcrumb-separator"
            >
              /
            </span>
          </span>
          <span>
            <span
              class="ant-breadcrumb-link"
            >
              Settings
            </span>
            <span
              class="ant-breadcrumb-separator"
            >
              /
            </span>
          </span>
        </div>
        <div
          class="ant-page-header site-page-header ant-page-header-ghost"
        >
          <div
            class="ant-page-header-heading"
          >
            <div
              class="ant-page-header-heading-left"
            >
              <div
                class="ant-page-header-back"
              >
                <div
                  aria-label="Back"
                  class="ant-page-header-back-button"
                  role="button"
                  style="border: 0px; background: transparent; padding: 0px; line-height: inherit; display: inline-block;"
                  tabindex="0"
                >
                  <span
                    aria-label="arrow-left"
                    class="anticon anticon-arrow-left"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="arrow-left"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 000 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
                      />
                    </svg>
                  </span>
                </div>
              </div>
              <span
                class="ant-page-header-heading-title"
                title="Settings"
              >
                Settings
              </span>
            </div>
          </div>
        </div>
        <div>
          <div
            class="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-left"
            role="separator"
          >
            <span
              class="ant-divider-inner-text"
            >
              <span
                aria-label="flag"
                class="anticon anticon-flag"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="flag"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M184 232h368v336H184z"
                    fill="#f8f0ff"
                  />
                  <path
                    d="M624 632c0 4.4-3.6 8-8 8H504v73h336V377H624v255z"
                    fill="#f8f0ff"
                  />
                  <path
                    d="M880 305H624V192c0-17.7-14.3-32-32-32H184v-40c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v784c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V640h248v113c0 17.7 14.3 32 32 32h416c17.7 0 32-14.3 32-32V337c0-17.7-14.3-32-32-32zM184 568V232h368v336H184zm656 145H504v-73h112c4.4 0 8-3.6 8-8V377h216v336z"
                    fill="#B398DB"
                  />
                </svg>
              </span>
               Nationalities
            </span>
          </div>
          <div
            class="ant-list ant-list-split"
          >
            <div
              class="ant-spin-nested-loading"
            >
              <div
                class="ant-spin-container"
              >
                <ul
                  class="ant-list-items"
                >
                  <li
                    class="ant-list-item ant-list-item-no-flex"
                  >
                    <button
                      aria-checked="true"
                      class="ant-switch ant-switch-checked"
                      role="switch"
                      type="button"
                    >
                      <div
                        class="ant-switch-handle"
                      />
                      <span
                        class="ant-switch-inner"
                      />
                    </button>
                     
                    <img
                      alt="CH Flag"
                      height="24"
                      role="img"
                      src="https://cdn.jsdelivr.net/gh/madebybowtie/FlagKit@2.2/Assets/SVG/CH.svg"
                      width="24"
                    />
                     
                    <span
                      class="ant-typography"
                    >
                      Switzerland
                    </span>
                  </li>
                  <li
                    class="ant-list-item ant-list-item-no-flex"
                  >
                    <button
                      aria-checked="true"
                      class="ant-switch ant-switch-checked"
                      role="switch"
                      type="button"
                    >
                      <div
                        class="ant-switch-handle"
                      />
                      <span
                        class="ant-switch-inner"
                      />
                    </button>
                     
                    <img
                      alt="ES Flag"
                      height="24"
                      role="img"
                      src="https://cdn.jsdelivr.net/gh/madebybowtie/FlagKit@2.2/Assets/SVG/ES.svg"
                      width="24"
                    />
                     
                    <span
                      class="ant-typography"
                    >
                      Spain
                    </span>
                  </li>
                  <li
                    class="ant-list-item ant-list-item-no-flex"
                  >
                    <button
                      aria-checked="true"
                      class="ant-switch ant-switch-checked"
                      role="switch"
                      type="button"
                    >
                      <div
                        class="ant-switch-handle"
                      />
                      <span
                        class="ant-switch-inner"
                      />
                    </button>
                     
                    <img
                      alt="FR Flag"
                      height="24"
                      role="img"
                      src="https://cdn.jsdelivr.net/gh/madebybowtie/FlagKit@2.2/Assets/SVG/FR.svg"
                      width="24"
                    />
                     
                    <span
                      class="ant-typography"
                    >
                      France
                    </span>
                  </li>
                  <li
                    class="ant-list-item ant-list-item-no-flex"
                  >
                    <button
                      aria-checked="true"
                      class="ant-switch ant-switch-checked"
                      role="switch"
                      type="button"
                    >
                      <div
                        class="ant-switch-handle"
                      />
                      <span
                        class="ant-switch-inner"
                      />
                    </button>
                     
                    <img
                      alt="GB Flag"
                      height="24"
                      role="img"
                      src="https://cdn.jsdelivr.net/gh/madebybowtie/FlagKit@2.2/Assets/SVG/GB.svg"
                      width="24"
                    />
                     
                    <span
                      class="ant-typography"
                    >
                      United Kingdom
                    </span>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
